e3sm_to_cmip/util.py

```diff
@@ -306,195 +359,389 @@ def print_var_info(handlers, freq=None, inpath=None, tables=None, outpath=None):
                 if not has_vars:
                     continue
                 messages.append(msg)
-    
-    
+
     if outpath is not None:
-        with open(outpath, 'w') as outstream:
+        with open(outpath, "w") as outstream:
             yaml.dump(messages, outstream)
     else:
         pprint(messages)
+
+
 # ------------------------------------------------------------------
 
-def get_table(table, variable, freq, tables):
+
+def _load_handlers(
+    handlers_path: str,
+    tables_path: str,
+    var_list: List[str],
+    freq: str = "mon",
+    realm: str = "atm",
+    simple: bool = False,
+) -> List[Dict[str, Any]]:
+    """Loads default and/or complex handlers based on a list of variable names.
```

Comment:
Semantics.  Anything that "loads default and/or complex handlers" simply "loads handlers".   Only make a distinction where it is needed.  I assume "complex" means that multiple variables are involved.  But any process capable of managing a "complex" handler should have no problem managing a simpler "default" handler.  Why maintain the distinction in separate codes?

Reply:
Sterling defined the handlers loaded from the `/cmor_handlers` directory as "more complex", so I carried that name over.

https://github.com/E3SM-Project/e3sm_to_cmip/blob/cee5c228f6085b0afd4a4c4696f0c947b1514f50/e3sm_to_cmip/util.py#L427-L428

I must have removed the comments that defined what default handlers (handlers loaded via `default_handlers_info.yaml`), and complex handlers (Python modules defined in `/cmor_handlers())`.

I'll add a clear description of the differences between these two in the docstring.

> But any process capable of managing a "complex" handler should have no problem managing a simpler "default" handler. Why maintain the distinction in separate codes?

Processing default and complex handlers involves different logic, so they are split into separate functions.

https://github.com/E3SM-Project/e3sm_to_cmip/blob/5059810b2cdf98031cafc1c25a6c4056ea5e5bed/e3sm_to_cmip/util.py#L478-L529

https://github.com/E3SM-Project/e3sm_to_cmip/blob/5059810b2cdf98031cafc1c25a6c4056ea5e5bed/e3sm_to_cmip/util.py#L532-L590

Reply:
I updated the function names for `_get_default_handlers()` to `_get_handlers_from_yaml()` and `_get_complex_handlers()` to `_get_handlers_from_modules()`.

All variable handlers are referred to as just "handlers", rather than "default handlers" or "complex handlers".

https://github.com/E3SM-Project/e3sm_to_cmip/blob/343ece4724fc5cea8a7cfc9f6b2d482c6bda5e4b/e3sm_to_cmip/util.py#L453-L471

https://github.com/E3SM-Project/e3sm_to_cmip/blob/343ece4724fc5cea8a7cfc9f6b2d482c6bda5e4b/e3sm_to_cmip/util.py#L474-L503

https://github.com/E3SM-Project/e3sm_to_cmip/blob/343ece4724fc5cea8a7cfc9f6b2d482c6bda5e4b/e3sm_to_cmip/util.py#L531-L574